zeroconf/__init__.py

```diff
@@ -2359,9 +2359,8 @@ def handle_response(self, msg: DNSIncoming) -> None:
         are held in the cache, and listeners are notified."""
         now = current_time_millis()
         for record in msg.answers:
-            expired = record.is_expired(now)
             if record in self.cache.entries():
-                if expired:
```

Comment:
I ran an experiment to test the performance of this change. I found no noticeable changes in performance due to this commit.

On a network with 170 devices, I ran a performance test. Every minute a request for broadcast is made. The Zeroconf ServiceBrowser is made to listen for incoming broadcasts. The ServiceBrowser was running on a separate thread while on the main thread an I/O-limited QNetworkRequest was made to include the effect of thread switching into my measurements. The experiment was ran on a MacBook Air 2015 (Intel Core i7-5650U) through Ubuntu 19.10 and Python 3.7. I left this running for exactly one hour, both before and after this single commit.

Before this commit, 139025204 (139M) calls were made to `DNSPointer.is_expired`, totalling 381.993s (just over 6 minutes).

After this commit, 140926403 (140M) calls were made to `DNSPointer.is_expired`, totalling 377.424s (just over 6 minutes).

I reckon that this falls within my measurement inaccuracy, so there is no noticeable change due to this commit in my experiment. Apparently `record` is in `self.cache.entries()` pretty much every time in my case.

However even though there was no performance benefit in practice, there is the possibility that there is a performance benefit for other cases. One could construe a test that resets the DNS cache every time to effectively measure start-up time for an application. I didn't take the effort to do this. In theory, it's easy to see that this change can only have benefits and no disadvantages. It's strictly doing less work, so I think this commit is still valid.